inc/Engine/License/API/Pricing.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace WP_Rocket\Engine\License\API;
+
+class Pricing {
+	/**
+	 * The pricing data object
+	 *
+	 * @var object
+	 */
+	private $pricing;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param object $pricing The pricing object.
+	 */
+	public function __construct( $pricing ) {
+		$this->pricing = $pricing;
+	}
+
+	/**
+	 * Gets the single license pricing data
+	 *
+	 * @return null|object
+	 */
+	public function get_single_pricing() {
+		return isset( $this->pricing->licenses->single ) ? $this->pricing->licenses->single : null;
+	}
+
+	/**
+	 * Gets the plus license pricing data
+	 *
+	 * @return null|object
+	 */
+	public function get_plus_pricing() {
+		return isset( $this->pricing->licenses->plus ) ? $this->pricing->licenses->plus : null;
+	}
+
+	/**
+	 * Gets the infinite license pricing data
+	 *
+	 * @return null|object
+	 */
+	public function get_infinite_pricing() {
+		return isset( $this->pricing->licenses->infinite ) ? $this->pricing->licenses->infinite : null;
+	}
+
+	/**
+	 * Gets the license renewal pricing data
+	 *
+	 * @return null|object
+	 */
+	public function get_renewals_data() {
+		return isset( $this->pricing->renewals ) ? $this->pricing->renewals : null;
+	}
+
+	/**
+	 * Gets the promotion data
+	 *
+	 * @return null|object
+	 */
+	public function get_promo_data() {
+		return isset( $this->pricing->promo ) ? $this->pricing->promo : null;
+	}
+
+	/**
+	 * Checks if a promotion is currently active
+	 *
+	 * @return boolean
+	 */
+	public function is_promo_active() {
+		$promo_data = $this->get_promo_data();
+
+		if ( is_null( $promo_data ) ) {
+			return false;
+		}
+
+		if ( ! isset( $promo_data->start_date, $promo_data->end_date ) ) {
+			return false;
+		}
+
+		$current_time = time();
+
+		return (
+			$promo_data->start_date < $current_time
+			&&
+			$promo_data->end_date > $current_time
+		);
+	}
+
+	/**
+	 * Gets promotion end date
+	 *
+	 * @return int
+	 */
+	public function get_promo_end() {
+		return isset( $this->get_promo_data()->end_date ) ? $this->get_promo_data()->end_date : 0;
```

Comment:
This one is calling `get_promo_data()` twice. 

```suggestion
		$promo = $this->get_promo_data();
		if ( is_null( $promo ) || ! isset( $promo->end_date ) ) {
			return 0;
		}
		
		return (int) $promo->end_date;
```